### PR TITLE
*/*: fix VariableShadowed

### PR DESCRIPTION
**Pull request by arthurzam**
As a result of new pkgcheck check https://github.com/pkgcore/pkgcheck/pull/623, it captures global assign shadows, and reports them. So let's fix all of them 😄 . The only once I gave up were:

```
sci-mathematics/fricas
  VariableShadowed: version 1.3.8-r1: possible variable 'n' is shadowed, on lines: 29, 45
  VariableShadowed: version 1.3.9: possible variable 'n' is shadowed, on lines: 30, 46
```

<details>
  <summary>Original Results for ::gentoo</summary>
  
  ```
  app-editors/neovim
    VariableShadowed: version 0.9.0-r1: possible variable 'REQUIRED_USE' is shadowed, on lines: 26, 29
    VariableShadowed: version 0.9.1: possible variable 'REQUIRED_USE' is shadowed, on lines: 26, 29
    VariableShadowed: version 0.9.2: possible variable 'REQUIRED_USE' is shadowed, on lines: 26, 29
    VariableShadowed: version 9999: possible variable 'REQUIRED_USE' is shadowed, on lines: 26, 29
  
  app-emulation/libguestfs
    VariableShadowed: version 1.48.4: possible variable 'BDEPEND' is shadowed, on lines: 110, 117
    VariableShadowed: version 1.48.6: possible variable 'BDEPEND' is shadowed, on lines: 110, 117
  
  dev-db/sqldeveloper
    VariableShadowed: version 23.1.0.097.1607: possible variable 'S' is shadowed, on lines: 11, 31
  
  dev-java/fop
    VariableShadowed: version 2.8-r1: possible variable 'BDEPEND' is shadowed, on lines: 50, 62
  
  dev-lang/luajit
    VariableShadowed: version 2.1.0_beta3_p20220127-r2: possible variable 'SRC_URI' is shadowed, on lines: 23, 24
    VariableShadowed: version 2.1.0_beta3_p20220613: possible variable 'SRC_URI' is shadowed, on lines: 23, 24
  
  dev-libs/gmp
    VariableShadowed: version 6.2.1-r5: possible variable 'MANUAL_PV' is shadowed, on lines: 11, 12
    VariableShadowed: version 6.3.0: possible variable 'MANUAL_PV' is shadowed, on lines: 11, 12
  
  dev-libs/libcgroup
    VariableShadowed: version 0.41-r6: possible variable 'PATCHES' is shadowed, on lines: 30, 45
  
  dev-ml/ppx_deriving
    VariableShadowed: version 5.2: possible variable 'DEPEND' is shadowed, on lines: 18, 26
    VariableShadowed: version 5.2.1: possible variable 'DEPEND' is shadowed, on lines: 18, 28
  
  dev-ml/stdcompat
    VariableShadowed: version 19: possible variable 'SLOT' is shadowed, on lines: 11, 13
  
  dev-perl/DBD-MariaDB
    VariableShadowed: version 1.210.0-r1: possible variable 'DEPEND' is shadowed, on lines: 23, 28
    VariableShadowed: version 1.220.0: possible variable 'DEPEND' is shadowed, on lines: 23, 28
  
  dev-php/pecl-translit
    VariableShadowed: version 0.7.1: possible variable 'PHP_EXT_NAME' is shadowed, on lines: 6, 11
  
  dev-php/pecl-eio
    VariableShadowed: version 3.0.0_rc2: possible variable 'LICENSE' is shadowed, on lines: 19, 22
    VariableShadowed: version 3.0.0_rc4: possible variable 'LICENSE' is shadowed, on lines: 19, 22
    VariableShadowed: version 3.1.0_rc1: possible variable 'LICENSE' is shadowed, on lines: 19, 22
  
  dev-python/keystoneauth1
    VariableShadowed: version 5.2.1: possible variable 'HOMEPAGE' is shadowed, on lines: 12, 13
    VariableShadowed: version 5.3.0: possible variable 'HOMEPAGE' is shadowed, on lines: 12, 13
  
  games-puzzle/pauker
    VariableShadowed: version 1.8-r3: possible variable 'BDEPEND' is shadowed, on lines: 19, 30
  
  gnome-base/gnome-control-center
    VariableShadowed: version 44.3: possible variable 'BDEPEND' is shadowed, on lines: 25, 120
    VariableShadowed: version 45_rc: possible variable 'BDEPEND' is shadowed, on lines: 25, 119
  
  media-gfx/viewer
    VariableShadowed: version 0.8.0-r1: possible variable 'SRC_URI' is shadowed, on lines: 10, 11
  
  media-libs/suil
    VariableShadowed: version 0.10.18-r1: possible variable 'BDEPEND' is shadowed, on lines: 18, 47
  
  media-libs/rubberband
    VariableShadowed: version 3.3.0-r1: possible variable 'BDEPEND' is shadowed, on lines: 17, 34
  
  media-libs/gstreamer-editing-services
    VariableShadowed: version 1.20.3: possible variable 'RESTRICT' is shadowed, on lines: 19, 34
    VariableShadowed: version 1.20.4: possible variable 'RESTRICT' is shadowed, on lines: 19, 34
    VariableShadowed: version 1.20.5: possible variable 'RESTRICT' is shadowed, on lines: 19, 34
    VariableShadowed: version 1.20.6: possible variable 'RESTRICT' is shadowed, on lines: 19, 34
    VariableShadowed: version 1.22.3: possible variable 'RESTRICT' is shadowed, on lines: 19, 34
  
  media-sound/cmus
    VariableShadowed: version 2.10.0-r1: possible variable 'REQUIRED_USE' is shadowed, on lines: 25, 64
    VariableShadowed: version 9999: possible variable 'REQUIRED_USE' is shadowed, on lines: 25, 64
  
  media-sound/qpaeq
    VariableShadowed: version 16.1: possible variable 'S' is shadowed, on lines: 24, 44
  
  media-sound/zynaddsubfx
    VariableShadowed: version 3.0.6-r1: possible variable 'BDEPEND' is shadowed, on lines: 19, 41
  
  media-sound/pulseaudio-daemon
    VariableShadowed: version 16.1-r3: possible variable 'S' is shadowed, on lines: 22, 161
    VariableShadowed: version 16.1-r6: possible variable 'S' is shadowed, on lines: 22, 161
    VariableShadowed: version 16.1-r7: possible variable 'S' is shadowed, on lines: 22, 160
  
  net-analyzer/nagstamon
    VariableShadowed: version 3.10.1: possible variable 'DESCRIPTION' is shadowed, on lines: 15, 16
    VariableShadowed: version 3.12.0: possible variable 'DESCRIPTION' is shadowed, on lines: 16, 17
  
  net-p2p/nicotine+
    VariableShadowed: version 3.2.8: possible variable 'DEPEND' is shadowed, on lines: 20, 27
    VariableShadowed: version 3.2.9: possible variable 'DEPEND' is shadowed, on lines: 20, 27
    VariableShadowed: version 3.2.9-r1: possible variable 'DEPEND' is shadowed, on lines: 21, 28
  
  sci-chemistry/molequeue
    VariableShadowed: version 0.9.0-r1: possible variable 'RESTRICT' is shadowed, on lines: 18, 42
  
  net-nds/openldap
    VariableShadowed: version 2.5.14: possible variable 'RESTRICT' is shadowed, on lines: 36, 38
    VariableShadowed: version 2.5.16: possible variable 'RESTRICT' is shadowed, on lines: 36, 38
    VariableShadowed: version 2.6.3-r7: possible variable 'RESTRICT' is shadowed, on lines: 36, 38
    VariableShadowed: version 2.6.4-r1: possible variable 'RESTRICT' is shadowed, on lines: 36, 38
    VariableShadowed: version 2.6.4-r2: possible variable 'RESTRICT' is shadowed, on lines: 37, 39
  
  sci-mathematics/fricas
    VariableShadowed: version 1.3.8-r1: possible variable 'n' is shadowed, on lines: 29, 45
    VariableShadowed: version 1.3.9: possible variable 'n' is shadowed, on lines: 30, 46
  
  sys-block/nbd
    VariableShadowed: version 3.24: possible variable 'BDEPEND' is shadowed, on lines: 23, 32
  
  sys-libs/ncurses-compat
    VariableShadowed: version 6.4_p20230401: possible variable 'SRC_URI' is shadowed, on lines: 19, 22
  
  sys-fs/btrfs-progs
    VariableShadowed: version 6.3.1: possible variable 'REQUIRED_USE' is shadowed, on lines: 36, 86
    VariableShadowed: version 6.3.2: possible variable 'REQUIRED_USE' is shadowed, on lines: 35, 85
    VariableShadowed: version 6.3.3: possible variable 'REQUIRED_USE' is shadowed, on lines: 35, 85
    VariableShadowed: version 6.5: possible variable 'REQUIRED_USE' is shadowed, on lines: 35, 85
    VariableShadowed: version 6.5.1: possible variable 'REQUIRED_USE' is shadowed, on lines: 35, 85
    VariableShadowed: version 9999: possible variable 'REQUIRED_USE' is shadowed, on lines: 35, 85
  
  www-client/vivaldi-snapshot
    VariableShadowed: version 6.3.3119.4: possible variable 'RE' is shadowed, on lines: 104, 105, 106, 107
    VariableShadowed: version 6.3.3120.3: possible variable 'RE' is shadowed, on lines: 104, 105, 106, 107
  
  www-client/vivaldi
    VariableShadowed: version 6.2.3105.47: possible variable 'RE' is shadowed, on lines: 104, 105, 106, 107
    VariableShadowed: version 6.2.3105.48: possible variable 'RE' is shadowed, on lines: 104, 105, 106, 107
  ```
</details>

### CODE REVIEW

File: dev-ml/ppx_deriving/ppx_deriving-5.2.1.ebuild [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright 1999-2022 Gentoo Authors
+# Copyright 1999-2023 Gentoo Authors
```
Comment (thesamesam) [inline dev-ml/ppx_deriving/ppx_deriving-5.2.1.ebuild:1]: This one needs git mv to fix broken DEPEND, or we won't get rebuilds

File: dev-ml/ppx_deriving/ppx_deriving-5.2.ebuild [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Copyright 1999-2021 Gentoo Authors
+# Copyright 1999-2023 Gentoo Authors
```
Comment (thesamesam) [inline dev-ml/ppx_deriving/ppx_deriving-5.2.ebuild:1]: git mv

File: media-libs/rubberband/rubberband-3.3.0-r1.ebuild [hunk 3]
```diff
@@ -26,13 +24,13 @@ CDEPEND="
 	programs? ( media-libs/libsndfile[${MULTILIB_USEDEP}] )
 	vamp? ( media-libs/vamp-plugin-sdk[${MULTILIB_USEDEP}] )
 "
-RDEPEND="
-	${CDEPEND}
+RDEPEND="${DEPEND}
```
Comment (thesamesam) [inline media-libs/rubberband/rubberband-3.3.0-r1.ebuild:27]:
```suggestion
RDEPEND="
	${DEPEND}
```

File: www-client/vivaldi-snapshot/vivaldi-snapshot-6.3.3119.4.ebuild [hunk 4]
```diff
@@ -101,10 +101,11 @@ fi
 KEYWORDS="-* ~amd64 ~arm ~arm64"
 VIVALDI_BASE_URI="https://downloads.vivaldi.com/${VIVALDI_PN#vivaldi-}/${VIVALDI_PN}_${PV%_p*}-${DEB_REV}_"
 
-RE="\bamd64\b"; [[ ${KEYWORDS} =~ ${RE} ]] && SRC_URI+=" amd64? ( ${VIVALDI_BASE_URI}amd64.deb )"
```
Comment (thesamesam) [inline www-client/vivaldi-snapshot/vivaldi-snapshot-6.3.3119.4.ebuild:104]: that is uh, very creative

File: www-client/vivaldi-snapshot/vivaldi-snapshot-6.3.3119.4.ebuild [hunk 5]
```diff
@@ -101,10 +101,11 @@ fi
...
-RE="\barm\b"; [[ ${KEYWORDS} =~ ${RE} ]] && SRC_URI+=" arm? ( ${VIVALDI_BASE_URI}armhf.deb )"
-RE="\barm64\b"; [[ ${KEYWORDS} =~ ${RE} ]] && SRC_URI+=" arm64? ( ${VIVALDI_BASE_URI}arm64.deb )"
-RE="\bx86\b"; [[ ${KEYWORDS} =~ ${RE} ]] && SRC_URI+=" x86? ( ${VIVALDI_BASE_URI}i386.deb )"
+SRC_URI="
```
Comment (thesamesam) [inline www-client/vivaldi-snapshot/vivaldi-snapshot-6.3.3119.4.ebuild:104]: This isn't quite the same, as the idea is to check if it's in KEYWORDS, and allow dropping from KEYWORDS if the packageh as no binary available right now.

But I have no idea how often that happens or why this bit is there to begin with..

**Comment by thesamesam** [comment]
(please rebase to fix CI)

File: www-client/vivaldi-snapshot/vivaldi-snapshot-6.3.3119.4.ebuild (hunk 5 above)
Comment (arthurzam) [reply www-client/vivaldi-snapshot/vivaldi-snapshot-6.3.3119.4.ebuild:104]: I think it is more readable (and faster in global scope) to just list them close one to another. As a result I think the new variant is better.

File: media-libs/gstreamer-editing-services/gstreamer-editing-services-1.22.3.ebuild [hunk 6]
```diff
@@ -9,14 +9,16 @@ inherit meson python-r1
...
 
 LICENSE="LGPL-2+"
 SLOT="1.0"
 KEYWORDS="~amd64 ~x86"
 
 IUSE="+introspection test"
 REQUIRED_USE="${PYTHON_REQUIRED_USE}"
-RESTRICT="!test? ( test )"
+# Some tests are failing
+RESTRICT="test !test? ( test )"
```
Comment (SoapGentoo) [inline media-libs/gstreamer-editing-services/gstreamer-editing-services-1.22.3.ebuild:21]: ditto

File: net-nds/openldap/openldap-2.5.14.ebuild [hunk 7]
```diff
@@ -35,7 +35,6 @@ IUSE_CONTRIB="${IUSE_CONTRIB} cxx"
 IUSE="systemd ${IUSE_DAEMON} ${IUSE_BACKEND} ${IUSE_OVERLAY} ${IUSE_OPTIONAL} ${IUSE_CONTRIB}"
 RESTRICT="!test? ( test )"
 
-RESTRICT="!test? ( test )"
 REQUIRED_USE="cxx? ( sasl )
```
Comment (SoapGentoo) [inline net-nds/openldap/openldap-2.5.14.ebuild:38]: `REQUIRED_USE` should come before `RESTRICT` and should be in the same block

File: www-client/vivaldi-snapshot/vivaldi-snapshot-6.3.3119.4.ebuild (hunk 5 above)
Comment (SoapGentoo) [reply www-client/vivaldi-snapshot/vivaldi-snapshot-6.3.3119.4.ebuild:104]: yeah, I prefer arthur's new code, way better

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

**Comment by arthurzam** [comment]
Merged

**Pull request closed without merging**